pkg/datapath/loader/compile.go

```diff
@@ -192,6 +194,17 @@ func compileAndLink(ctx context.Context, prog *progInfo, dir *directoryInfo, deb
 	return err
 }
 
+// GetHardwareName returns hardware name on this host as 'uname -m'.
+func GetHardwareName() (string, error) {
+	var unameBuf unix.Utsname
+	err := unix.Uname(&unameBuf)
+	if err != nil {
+		log.WithError(err).Fatal("Get machine hardware name error!")
```

Comment:
Please do not fatal here, return the error to the caller.

Reply:
Corrected by https://github.com/cilium/cilium/pull/9900/commits/73ea3c61dac32e5f8c9694f157070a40223a3d4e